.github/scripts/promote_branch.sh

```diff
@@ -0,0 +1,160 @@
+    echo
+    echo "  --target-branch:    The name of the branch to be promoted. Options are staging"
+    echo "                      and production."
+    echo "  --force-to-staging: If passed with value true, allow promotion to staging even"
+    echo "                      if staging and production differ."
+    echo "  --override:         If passed with value true, allow promotion to production"
+    echo "                      even if the change has not been in staging for one week."
+    echo "  --debug:            For debugging purposes, do everything except for the git"
+    echo "                      push and deleting the temp repo."
+    echo
+    echo "  --target-branch has to be specified."
+}
+
+check_if_branch_differs() {
+    ACTUAL_DIFFERENT_LINES=$(git diff --numstat origin/$1 | wc -l)
+    if [ $ACTUAL_DIFFERENT_LINES -ne 0 ] ; then
+        echo "Lines differ in branch $1"
+        echo "Actual differing lines: $(git diff --numstat origin/$1)"
+        exit 1
+    fi
+}
+
+check_if_any_commits_in_last_week() {
+    NEW_COMMITS=$(git log --oneline --since="$(date --date="7 days ago" +%Y-%m-%d)" | wc -l)
+    if [ $NEW_COMMITS -ne 0 ] ; then
+        echo "There are commits in staging that are less than a week old. Blocking promotion to promotion"
+        echo "Commits less than a week old: $(git log --oneline --since="$(date --date="7 days ago" +%Y-%m-%d)")"
+        exit 1
+    fi
+}
+
+if [ -z "${TARGET_BRANCH}" ]; then
+    echo -e "Error: missing 'target-branch' argument\n"
+    print_help
+    exit 1
+fi
+if [ "${TARGET_BRANCH}" == "staging" ]; then
+    SOURCE_BRANCH="development"
+elif [ "${TARGET_BRANCH}" == "production" ]; then
+    SOURCE_BRANCH="staging"
+else
+    echo "Invalid target-branch. Only 'staging' and 'production' are allowed"
+    print_help
+    exit 1
+fi
+if [ -z "${GITHUB_TOKEN}" ]; then
+    echo -e "Error: missing 'GITHUB_TOKEN' environment variable\n"
+    print_help
+    exit 1
+fi
+
+# Personal access token with appropriate permissions
+token="${GITHUB_TOKEN}"
+
+# Clone the repository
+tmpDir=$(mktemp -d)
+releaseServiceCatalogDir=${tmpDir}/release-service-catalog
+mkdir -p ${releaseServiceCatalogDir}
+
+echo -e "---\nPromoting release-service-catalog ${SOURCE_BRANCH} to ${TARGET_BRANCH}\n---\n"
+
+git clone "git@github.com:$ORG/$REPO.git" ${releaseServiceCatalogDir}
+cd ${releaseServiceCatalogDir}
+
+# A change cannot go into production if the changes in staging are less than a week old
+if [[ "${TARGET_BRANCH}" == "production" && "${OVERRIDE}" != "true" ]] ; then
+    git checkout origin/staging
+    check_if_any_commits_in_last_week
+fi
+
+# A change cannot go into staging if staging and production differ
+if [[ "${TARGET_BRANCH}" == "staging" && "${FORCE_TO_STAGING}" != "true" ]] ; then
+    git checkout origin/staging
+    update_taskGitRevision_in_pipelines production
+    check_if_branch_differs production
+fi
+
+description="Included PRs:\r\n"
```

Comment:
What's the `\r` for? I think it's redundant. Probably came from the other script. But let's remove it.

Reply:
Yep, removed